src/main/kotlin/com/cognifide/gradle/aem/instance/tail/Blacklist.kt

```diff
@@ -0,0 +1,37 @@
+package com.cognifide.gradle.aem.instance.tail
+
+import com.cognifide.gradle.aem.common.file.FileOperations
+
+class Blacklist(
+    private val filters: List<(Log) -> Boolean> = emptyList(),
+    private val blacklists: List<String> = emptyList()
+) {
+
+    private val parser = Parser()
+
+    private val blacklist = loadBlacklists()
+
+    private fun loadBlacklists(): Map<String, Log> {
+        return (loadDefaultBlacklists(TailOptions.BLACKLIST_FILES_DEFAULT) +
+                loadConfiguredBlacklists(blacklists))
+                .flatten().map { it.messageChecksum to it }.toMap()
```

Comment:
you are comparing checksums of stored incidents / blacklisted with incoming ones...
isn't it too overengineered?

I am thinking that maybe these wildcard patterns on `* org.osgi.framework.ServiceException: Service factory returned null*` will be good enough

Reply:
and the blacklist file content will be set of wildcard rules or regex rules, one for each line

Reply:
basing on checksum could be problematic because of line endings (VCS/Git) could change them so that checksum will also change

Reply:
what is more such complex blacklists are very unstable because any change on stack trace on aem runtime will make them out of date